docs/src/main/sphinx/develop/client-protocol.rst

```diff
@@ -0,0 +1,257 @@
+      query line number and column number where the failure was detected.
+  * - ``warnings``
+    - ``List<TrinoWarning>``
+    - A usually-empty list of warnings.
+  * - ``statementStats``
+    - ``StatementStats``
+    - A class containing statistics about the query execution.  Of
+      particular interest is ``StatementStats.rootStage``, of type
+      ``StageStats``, providing statistics on the execution of each of
+      the stages of query processing.
+
+Client request headers
+======================
+
+This table lists all supported client request headers.  Many of the
+headers can are updated in the client by response headers, and supplied
+in subsequent requests, just like browser cookies.
+
+.. list-table:: Client request headers
+  :widths: 30, 50
+  :header-rows: 1
+
+  * - Header Name
+    - Description
+  * - ``X-Trino-User``
+    - Specifies the session user; must be supplied with every
+      request to ``/v1/statement``.
+  * - ``X-Trino-Source``
+    - For reporting purposes, this supplies the name of the software
+      that submitted the query.
+  * - ``X-Trino-Catalog``
+    - The catalog to be used when running the query.  Set by response
+      header ``X-Trino-Set-Catalog``.
+  * - ``X-Trino-Schema``
+    - The schema to be used when running the query.  Set by response
+      header ``X-Trino-Set-Schema``.
+  * - ``X-Trino-Time-Zone``
+    - The timezone to be used when running the query, which by default
+      is the timezone of the Trino engine.
+  * - ``X-Trino-Language``
+    - The language to be used when running the query and formatting
+      results.  The language of the session can be set on a
+      per-query basis using the ``X-Trino-Language`` HTTP header,
+      or via the ``TrinoConnection.setLocale(Locale)`` method in
+      the JDBC driver.
+  * - ``X-Trino-Trace-Token``
+    - Supplies a trace token to the Trino engine to help identify
+      log lines that originate with this query request.
+  * - ``X-Trino-Session``
+    - Supplies a comma-separated list of name=value pairs as session
+      properties.  When the Trino client run a
+      ``SET SESSION name=value`` query, the name=value pair
+      is returned in the ``X-Set-Trino-Session`` response header,
+      and added to the client's list of session properties.
+      If the response header ``X-Trino-Clear-Session`` is returned,
+      its value is the name of a session property that will be
+      removed from the client's accumulated list.
+  * - ``X-Trino-Role``
+    - Sets the catalog role that will be used in this request.
+      Set by response header ``X-Trino-Set-Role``.
+  * - ``X-Trino-Prepared-Statement``
+    - A comma-separated list of the name=value pairs, where the
+      names are names of previously prepared SQL statements, and
+      the values are keys that identify the executable form of the
+      named prepared statements.
+  * - ``X-Trino-Transaction-Id``
+    - The transaction ID to be used when running the query.  Set
```

Comment:
The transaction ID to use for query processing.

Reply:
Changed.